packages/lit-html/src/directives/cache.ts

```diff
@@ -31,30 +42,49 @@ import {detachNodePart, restoreNodePart, NodePartState} from '../parts.js';
  */
 export const cache = directive(
   class extends Directive {
-    templateCache = new WeakMap<TemplateStringsArray, NodePartState>();
+    templateCache = new WeakMap<TemplateStringsArray, CachedPart>();
     value?: TemplateResult;
 
     render(v: unknown) {
-      return v;
+      // Return an array of the value to induce lit-html to create a NodePart
+      // for the value that we can move into the cache.
+      return [v];
     }
 
-    update(part: NodePart, [v]: Parameters<this['render']>) {
+    update(containerPart: NodePart, [v]: Parameters<this['render']>) {
       // If the new value is not a TemplateResult from the same Template as the
       // previous value, move the nodes from the DOM into the cache.
       if (
         this.value !== undefined &&
         this.value.strings !== (v as TemplateResult).strings
       ) {
-        this.templateCache.set(this.value.strings, detachNodePart(part));
+        // This is always an array because we return [v] in render()
+        const partValue = getPartValue(containerPart) as Array<NodePart>;
+        const childPart = partValue.pop()!;
+        let cachedPart = this.templateCache.get(this.value.strings);
+        if (cachedPart === undefined) {
+          const fragment = new DocumentFragment();
+          const containerPart = render(nothing, fragment);
```

Comment:
Should use a different name for this so it's not conflated with the function argument name.

Reply:
done